$MirOS: ports/devel/glib/patches/patch-gmodule_gmodule-dl_c,v 1.1.7.1 2005/03/18 15:44:37 tg Exp $
$OpenBSD: patch-gmodule_gmodule-dl_c,v 1.1 2004/12/07 06:16:21 marcm Exp $
--- gmodule/gmodule-dl.c.orig	Fri Sep 12 19:55:25 2008
+++ gmodule/gmodule-dl.c	Thu Nov 25 12:24:45 2010
@@ -107,6 +107,7 @@ _g_module_open (const gchar *file_name,
 static gpointer
 _g_module_self (void)
 {
+#if !defined(__OpenBSD__) || !defined(RTLD_DEFAULT)
   gpointer handle;
   
   /* to query symbols from the program itself, special link options
@@ -118,6 +119,9 @@ _g_module_self (void)
     g_module_set_error (fetch_dlerror (TRUE));
   
   return handle;
+#else
+  return RTLD_DEFAULT;
+#endif
 }
 
 static void
